Src/Particle/AMReX_ParticleContainerI.H

```diff
@@ -796,26 +851,24 @@ ParticleContainer<NStructReal, NStructInt, NArrayReal, NArrayInt, Allocator>
     if (level >= static_cast<int>(m_particles.size()))
         return;
 
-    const BoxArray& fine = ParticleBoxArray(level + 1);
-    std::vector< std::pair<int,Box> > isects;
+    if (! m_particle_locator.isValid(GetParGDB())) m_particle_locator.build(GetParGDB());
+    m_particle_locator.setGeometry(GetParGDB());
+    AmrAssignGrid<DenseBinIteratorFactory<Box>> assign_grid = m_particle_locator.getGridAssignor();
+
+    auto ghost_offset = ghosts.numParticles();
     for(ParConstIterType pti(*this, level); pti.isValid(); ++pti)
     {
-        const auto& ptile = ParticlesAt(level, pti);
-        const auto src = ptile.getConstParticleTileData();
+        const auto& src_tile = ParticlesAt(level, pti);
+        int gid = pti.index();
 
-        auto& aos = ptile.GetArrayOfStructs();
-        for (int pindex = 0; pindex < ptile.numParticles(); ++pindex)
-        {
-            const IntVect& iv = Index(aos[pindex], level+1);
-            fine.intersections(Box(iv,iv),isects,true,nGrow);
-            if (isects.size() > 0)
-            {
-                SuperParticleType p = src.getSuperParticle(pindex);
-                p.id() = GhostParticleID;
-                ghosts.push_back(p);
-            }
-        }
+        auto np = src_tile.numParticles();
+        ghosts.resize(ghost_offset+np);
+        ghost_offset += filterAndTransformParticles(ghosts, src_tile, AssignGridFilter(assign_grid,gid,level,nGrow), TransformerGhost(),0,ghost_offset);
+        ghosts.resize(ghost_offset);
+        Gpu::synchronize();
```

Comment:
`resize` implies a `sync`, so you don't need to do so explicitly here.

Reply:
Pushed update commit, removed extra resize and extra sync